### PR TITLE
Refactoring: Simplify adding new configuration properties

### PR DESCRIPTION
**Pull request by henrymercer**
This PR refactors the way we initialize and test configurations to make it easier to add new configuration properties.  It shouldn't have any user visible effects.  Specifically, adding a new configuration property should not require updating irrelevant tests.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: src/config-utils.ts [hunk 1]
```diff
@@ -391,39 +391,69 @@ export async function getRawLanguages(
...
+  repository: RepositoryNwo;
+  tempDir: string;
+  codeql: CodeQL;
+  workspacePath: string;
+  githubVersion: GitHubVersion;
+  apiDetails: api.GitHubApiCombinedDetails;
+  logger: Logger;
+}
+
+type GetDefaultConfigInputs = Omit<
```
Comment (angelapwen) [inline src/config-utils.ts:415]: TIL about the `Omit` type 🤯

File: src/config-utils.ts [hunk 2]
```diff
@@ -391,39 +391,69 @@ export async function getRawLanguages(
...
+  InitConfigInputs,
+  "configFile" | "configInput"
+>;
+
+type LoadConfigInputs = Omit<InitConfigInputs, "configInput"> & {
```
Comment (angelapwen) [inline src/config-utils.ts:420]: What does this `&` operator do here? Thought it represented bitwise AND but I can't really make sense of it in this context.

File: src/config-utils.ts (hunk 1 above)
Comment (angelapwen) [reply src/config-utils.ts:415]: We could use `Omit`/`Pick` to refactor the `StatusReportBase` and all those other interfaces later, perhaps...

**Review by angelapwen — APPROVED** [review]
Just a quick question 😄

File: src/config-utils.ts (hunk 2 above)
Comment (aeisenberg) [reply src/config-utils.ts:420]: These are [intersections](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#intersections). Though I wonder if this should be a union type (using `|`).

Comment (henrymercer) [reply src/config-utils.ts:420]: I believe we want intersections, i.e. we want the type to include the properties from both the left hand side and the right hand side.  For instance if we have (1) `{ a: 1 }` and (2) `{ a: 1 }`, then (1) and (2) would satisfy `{ a: int } | { b: string }` but only (2) would satisfy `{ a: int } & { b: string }`.

File: src/config-utils.ts (hunk 1 above)
Comment (henrymercer) [reply src/config-utils.ts:415]: There are quite a few of these nice types :) https://www.typescriptlang.org/docs/handbook/utility-types.html

**Pull request merged**